src/server.rs

```diff
@@ -354,6 +356,7 @@ where
 pub struct Server<T> {
     connection: Connection,
     imp: Arc<T>,
+    bus_name_suffix: String,
```

Comment:
I think it may be simpler to just store the entire bus name and return &str from `bus_name`.